daemon/containerd/progress.go

```diff
@@ -0,0 +1,153 @@
+package containerd
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd/content"
+	cerrdefs "github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/remotes"
+	"github.com/docker/docker/pkg/progress"
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+type updateProgressFunc func(ctx context.Context, ongoing *jobs, output progress.Output, start time.Time) error
+
+func showProgress(ctx context.Context, ongoing *jobs, out progress.Output, updateFunc updateProgressFunc) func() {
```

Comment:
Since `jobs` is only useful in conjunction with `showProgress` and vice versa, I feel that association should be formalized.

```suggestion
func (j *jobs) showProgress(ctx context.Context, out progress.Output, updateFunc updateProgressFunc) func() {
```